truenas_install/__main__.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding=utf-8 -*-
+import glob
+import json
+import logging
+import subprocess
+import sys
+
+logger = logging.getLogger(__name__)
+
+if __name__ == "__main__":
+    update = json.loads(sys.stdin.read())
+
+    root = update["root"]
+
+    subprocess.run(
+        [
+            "rsync", "-aRx",
+            "--exclude", "/data/factory-v1.db",
+            "--exclude", "/data/manifest.json",
+            "/etc/default/grub.d/truenas.cfg", "/etc/hostid", "/data", "/root",
+            f"{root}/",
+        ],
+        check=True,
+    )
+
+    undo = []
+    try:
+        subprocess.run(["mount", "-t", "proc", "none", f"{root}/proc"], check=True)
+        undo.append(["umount", f"{root}/proc"])
+
+        subprocess.run(["mount", "-t", "sysfs", "none", f"{root}/sys"], check=True)
+        undo.append(["umount", f"{root}/sys"])
+
+        for device in glob.glob("/dev/sd?*") + ["/dev/zfs"]:
```

Comment:
It could also be vd*

Wonder if there are others as well

Reply:
Here's list we currently have: 
https://github.com/truenas/truenas-installer/blob/master/usr/sbin/truenas-install#L139